spharpy/spherical.py

```diff
@@ -1,6 +1,431 @@
 import numpy as np
 import scipy.special as special
 import spharpy.special as _special
+import logging as logger
+from pyfar import Coordinates
+from functools import lru_cache
+from spharpy.samplings.helpers import calculate_sampling_weights
+
+
+class SphHarm:
+    def __init__(self, n_max, coords, basis_type='complex', channel_convention='acn', inv_transform_type=None,
+                 normalization='n3d'):
+        r"""
+        Create a spherical harmonics class for transformation between spherical harmonics and signals.
+
+        Parameters
+        ----------
+        n_max : int,
+            Maximum spherical harmonic order
+        coords : :doc:`pf.Coordinates <pyfar:classes/pyfar.coordinates>`
```

Comment:
If the inverse if 'quadrature' the coordinates must have weights. This should be mentioned. Do we need `spharpy.SamplingSphere` in this case instead of `pyfar.Coordinates`?

Reply:
the ``spharpy.SamplingSphere`` already contains an n_max, maybe we can use this ``n_max``. If we dont use the ``spharpy.SamplingSphere`` here, then why do we need it?

Reply:
I think it would be good to define n_max and the sampling scheme separately. There are some use cases where we don't want to transform at the maximum order